apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -136,4 +149,38 @@ export class ProcessSubscriber {
       transactionId: command.transactionId,
     });
   }
+
+  private async filterPreferredChannels(
+    organizationId: string,
+    environmentId: string,
+    subscriberId: string,
+    template: NotificationTemplateEntity
+  ): Promise<NotificationStepEntity[]> {
+    const buildCommand = BuildSubscriberPreferenceTemplateCommand.create({
+      organizationId: organizationId,
+      subscriberId: subscriberId,
+      environmentId: environmentId,
+      template,
+    });
+
+    const preference = (await this.buildSubscriberPreferenceTemplateUsecase.execute(buildCommand)).preference;
+
+    return template.steps.filter((step) => this.actionStep(step) || this.stepPreferred(preference, step));
+  }
+
+  private stepPreferred(preference: { enabled: boolean; channels: IPreferenceChannels }, step: NotificationStepEntity) {
+    const templatePreferred = preference.enabled;
+
+    const channelPreferred = Object.keys(preference.channels).some(
+      (channelKey) => channelKey === step.template.type && preference.channels[step.template.type]
+    );
+
+    return templatePreferred && channelPreferred;
+  }
+
+  private actionStep(step) {
+    const channels = ['in_app', 'email', 'sms', 'push', 'direct'];
```

Comment:
maybe worth using the ChanneTypeEnum for the values in this array

Reply:
Here I wanted to list only the channel type ones, while the ChanneTypeEnum have actions (digest,trigger) as well

Reply:
yes but if you do ChanneTypeEnum.SMS instead of 'sms' it ensures we do have same values here as in the enum...

Reply:
Oh you mean to use the values of the enum and not the enum as a list.
Well I definitely can but it will be partial at the moment because the direct and push are still missing there

Reply:
true but we could complete it later on 😄 better then none ;)

Reply:
yep pushed an update :)